webapp/src/modules/ui/map/sagas.js

```diff
@@ -1,16 +1,25 @@
-import { takeEvery, put } from 'redux-saga/effects'
-import { FETCH_PARCELS_REQUEST } from 'modules/parcels/actions'
-import { CHANGE_RANGE } from './actions'
+import { takeEvery, put, select } from 'redux-saga/effects'
+import { getParcels } from 'modules/parcels/reducer'
+import { buildCoordinate } from 'lib/utils'
+import { inBounds } from 'lib/parcelUtils'
+import { fetchParcelData, fetchParcels } from 'modules/parcels/actions'
+import { CHANGE_RANGE, HOVER_PARCEL } from './actions'
 
 export function* mapSaga() {
   yield takeEvery(CHANGE_RANGE, handleChangeRange)
+  yield takeEvery(HOVER_PARCEL, handleHoverParcel)
 }
 
 function* handleChangeRange(action) {
   const { nw, se } = action
-  yield put({
-    type: FETCH_PARCELS_REQUEST,
-    nw,
-    se
-  })
+  yield put(fetchParcels(nw, se))
+}
+
+function* handleHoverParcel(action) {
+  const { x, y } = action
+  if (!inBounds(x, y)) return
+  const parcels = yield select(getParcels)
+  const parcel = parcels[buildCoordinate(x, y)]
+  if (!parcel || !parcel.owner || parcel.data) return // only fetch data if the parcel is loaded and has an owner, but the land data has not been loaded yet
```

Comment:
Would it be too much of an overhead to re-fetch the data even if we already have it?
So you don't have to refresh to see changes in other parcels, no?

Reply:
Furthermore, doing some logging it seems like all parcels have an empty `data` object like this:
![image](https://user-images.githubusercontent.com/692077/35733597-4fecdd12-081e-11e8-9d80-648322dff908.png)

so my data it's not being fetched when I hover.
When I remove the `|| parcel.data)` to give it a try, I don't see the popup update:

![parcel](https://user-images.githubusercontent.com/692077/35733781-032980ce-081f-11e8-8785-08766a2b4234.gif)